pkg/agent/storage/https.go

```diff
@@ -246,7 +246,7 @@ func extractTarFiles(reader io.Reader, dest string) error {
 		}
 
 		// gosec G110
-		_, err := io.CopyN(newFile, tr, DEFAULT_MAX_DECOMPRESSION_SIZE)
+		_, err = io.CopyN(newFile, tr, DEFAULT_MAX_DECOMPRESSION_SIZE)
```

Comment:
maybe it is better to assign the error to a new env.
```suggestion
		_, ioErr := io.CopyN(newFile, tr, DEFAULT_MAX_DECOMPRESSION_SIZE)
```

Reply:
Changed.